lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,369 @@
+        vars:
+          - name: ansible_vmware_tools_vm_path
+        required: True
+      vm_username:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_vmware_tools_vm_username
+        required: True
+      vm_password:
+        description:
+          - VM password.
+        vars:
+          - name: ansible_vmware_tools_vm_password
+        required: True
+      exec_command_sleep_interval:
+        description:
+          - exec command sleep interval in seconds.
+        vars:
+          - name: ansible_vmware_tools_exec_command_sleep_interval
+        default: 5
+        type: integer
+      file_chunk_size:
+        description:
+          - File chunk size.
+          - "(Applicable when writing a file to disk, example: using the C(fetch) module.)"
+        vars:
+          - name: ansible_vmware_tools_file_chunk_size
+        default: 128
+        type: integer
+"""
+
+
+class Connection(ConnectionBase):
+    """VMware Tools Connection."""
+
+    transport = "vmware_tools"
+
+    @property
+    def connection_address(self):
+        """Read-only property holding the connection address."""
+        return self.get_option("connection_address")
+
+    @property
+    def connection_verify_ssl(self):
+        """Read-only property holding whether the connection should verify ssl."""
+        return self.get_option("connection_verify_ssl")
+
+    @property
+    def authManager(self):
+        """Guest Authentication Manager."""
+        return self._si.content.guestOperationsManager.authManager
+
+    @property
+    def fileManager(self):
+        """Guest File Manager."""
+        return self._si.content.guestOperationsManager.fileManager
+
+    @property
+    def processManager(self):
+        """Guest Process Manager."""
+        return self._si.content.guestOperationsManager.processManager
+
+    @property
+    def windowsGuest(self):
+        """Return if VM guest family is windows."""
+        return self.vm.guest.guestFamily == "windowsGuest"
+
+    def __init__(self, *args, **kwargs):
+        """init."""
+        super(Connection, self).__init__(*args, **kwargs)
+        if hasattr(self, "_shell") and self._shell.SHELL_FAMILY == "powershell":
+            self.module_implementation_preferences = (".ps1", ".exe", "")
+            self.become_methods = ["runas"]
+            self.allow_executable = False
+            self.has_pipelining = True
+            self.allow_extras = True
+
+    def _establish_connection(self):
+        connection_kwargs = {"host": self.connection_address, "user": self.get_option("connection_username"), "pwd": self.get_option("connection_password")}
+
+        if self.connection_verify_ssl:
+            connect = SmartConnect
+        else:
+            if self.get_option("connection_ignore_ssl_warnings"):
+                urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```

Comment:
Seems to me that you should just do this unconditionally.  What's the use case of setting `connection_verify_ssl=False`, but `connection_ignore_ssl_warnings=False`?

Reply:
It's two different, but similar things. `connection_verify_ssl=False` connects without verifying, while `connection_ignore_ssl_warnings=True` prevents the 'you really should SSL' warnings in the output. Maybe you want the warnings as kind of a nagging reminder to fix your SSL?